### ISSUE TITLE
the Order of Basis Functions for Hamiltonian_pred Matrix

### ISSUE THREAD

**Issue by Lllleosdu** [Xiaoxun-Gong/DeepH-E3#19]
Dear DeepH Development Team,

I have completed DeepH training using the data obtained from OpenMX. In the process of matrix diagonalization, I have learned that in the `DeepH-E3/inference_tools/sparse_calc.jl` script, a 364 * 13 * 2=9464 dimensional matrix was constructed using my target twisted bilayer BN (364 BN both s2p2d1,with SOC). However, I am not sure if the basis of this matrix are PAO used by OpenMX, and what the order of the basis is. Could you help me answer the question？

Thanks

**Comment by Xiaoxun-Gong** [comment]
Yes, if you construct your dataset using OpenMX, then all Hamiltonian matrices given by the trained neural network are under the same set of atomic orbital basis. The ordering of atomic orbital basis follow these rules:

First, they are ordered according to the order of atoms in the unit cell of the structure.
Second, within each atom, the atomic orbital basis with up spin is before those with down spin.
Third, within each spin, take s2p2d1 for example: orbitals number 1-2 are s orbitals with two different radial functions, orbitals number 3-8 are p orbitals with two different radial functions, orbitals number 9-13 are d orbitals with one radial function. The spherical harmonics follow OpenMX convention: Py, Pz, Px, Fz3, Fxz2, Fyz2, Fz(x2-y2), Fxyz, Fx(x2-3y2), Fy(3x2-y2).

**Comment by Lllleosdu** [comment]
Thanks very much for your detailed answer.

**Comment by Xiaoxun-Gong** [comment]
You are welcome. Thanks for your interest in DeepH-E3!

**Comment by jingzhe-alt** [comment]
Does it mean each Hamiltonian provided for the training should follow the orbital order as in openMX?  Since some code has different orbital order.

**Comment by Lllleosdu** [comment]
@jingzhe-alt I think DeepH(-E3) handles this in the preprocessing step. Data from different DFT codes are converted into a unified DeepH format before training.
For reference, the OpenMX basis/orbital order is listed in Table 9 of the OpenMX manual:
[OpenMX manual v3.9, Table 9](https://www.openmx-square.org/openmx_man3.9/openmx3.9.pdf)￼
